ai_economist/foundation/components/ubi.py

```diff
@@ -0,0 +1,246 @@
+import numpy as np
+
+from ai_economist.foundation.base.base_component import (
+    BaseComponent,
+    component_registry,
+)
+
+
+@component_registry.add
+class UBI(BaseComponent):
+    """Gives each mobile agent a constant amount of coin.
+
+    Args:
+        amt_min (float): Minimum UBI payment amount. Must be >= 0 (default).
+        amt_max (float): Maximum UBI payment amount. Default is 50 (arbitrary).
+        amt_disc (float): The interval separating discrete UBI payment amounts
+            that the planner can select. Default is 5 (arbitrary).
+        period (int): Length of a period in environment timesteps. UBI is
+            distributed at the end of each period. Must be > 0. Default is 100.
+    """
+
+    name = "UBI"
+    required_entities = ["Coin"]
+    agent_subclasses = ["BasicMobileAgent", "BasicPlanner"]
+
+    def __init__(
+        self,
+        *base_component_args,
+        amt_min=0,
+        amt_max=50,
+        amt_disc=5,
+        period=100,
+        **base_component_kwargs
+    ):
+        super().__init__(*base_component_args, **base_component_kwargs)
+
+        # Min and max UBI payment amounts
+        self.amt_min = amt_min
+        assert self.amt_min >= 0
+        self.amt_max = amt_max
+
+        # Size of each discrete step
+        self.amt_disc = amt_disc
+
+        # Discrete options for the amount of each UBI payment
+        self.disc_amts = np.arange(
+            self.amt_min, self.amt_max + self.amt_disc, self.amt_disc
+        )
+        self.disc_amts = self.disc_amts[self.disc_amts <= self.amt_max]
+        assert len(self.disc_amts) > 1
```

Comment:
needs amt_min < amt_max and amt_max - amt_min >= amt_disc ... could be more transparent to check for that instead?